Configuration/PyReleaseValidation/python/MatrixInjector.py

```diff
@@ -398,12 +398,28 @@ def prepare(self,mReader, directories, mode='init'):
                                     #t_input['DQMConfigCacheID']=t_second['ConfigCacheID']
                                 break
 
-            ## there is in fact only one acquisition era
-            #if len(set(chainDict['AcquisitionEra'].values()))==1:
-            #    print "setting only one acq"
+            # agreed changes for wm injection:
+            # - Campaign: *optional* string during creation. It will default to AcqEra value if possible.  
+            #             Otherwise it will be empty.
+            # - AcquisitionEra: *mandatory* string at request level during creation. *optional* string
+            #                   at task level during creation. "optional" during assignment.
+            # - ProcessingString: *mandatory* string at request level during creation. *optional* string
+            #                     at task level during creation. "optional" during assignment.
+            # - ProcessingVersion: *optional* during creation (default 1). *optional* during assignment.
+            # 
+            # Which requires following changes here:
+            #  - reset Global AcuisitionEra, ProcessingString to be the one in the first task
+            #  - and also Campaign to be always the same as the AcquisitionEra
+
             if acqEra:
-                chainDict['AcquisitionEra'] = chainDict['AcquisitionEra'].values()[0]
+                chainDict['AcquisitionEra'] = chainDict['AcquisitionEra'].values()[0] 
+                chainDict['ProcessingString'] = chainDict['ProcessingString'].values()[0]
+            else:
+                chainDict['AcquisitionEra'] = chainDict['nowmTasklist'][0]['AcquisitionEra']
+                chainDict['ProcessingString'] = chainDict['nowmTasklist'][0]['ProcessingString']
```

Comment:
@davidlange6  i think i might not get your point correctly.... to me, in both cases "AcquisitionEra" and "ProcessingString" are set here:
https://github.com/hengne/cmssw/blob/64ad67d1722b88f0038622e2ad873c6dc0be68f1/Configuration/PyReleaseValidation/python/MatrixInjector.py#L344-L351